### PR TITLE
feat(checkbox): Added mixin to customize checkbox touch dimension.

### PR DESCRIPTION
**Pull request by abhiomkar**
This new mixin allows customizing checkbox touch dimension. The default touch dimension is unchanged.

```sass
mdc-checkbox-touch-dimension(48px);
```

image

```sass
mdc-checkbox-touch-dimension(40px); // Default - current behaviour.
```

image

Refs #4694 

BREAKING CHANGE: Removed `$mdc-checkbox-ui-pct` sass variable from `MDCCheckbox`

### CODE REVIEW

**Comment by moog16** [comment]
This technically also fixes https://github.com/material-components/material-components-web/issues/4597

File: packages/mdc-checkbox/README.md [hunk 1]
```diff
@@ -122,6 +122,7 @@ Mixin | Description
 `mdc-checkbox-container-colors($unmarked-stroke-color, $unmarked-fill-color, $marked-stroke-color, $marked-fill-color, $generate-keyframes)` | Generates CSS classes to set and animate the stroke color and/or container fill color of a checkbox
 `mdc-checkbox-ink-color($color)` | Sets the ink color of the checked and indeterminate icons
 `mdc-checkbox-focus-indicator-color($color)` | Sets the color of the focus indicator
+`mdc-checkbox-touch-dimension($touch-dimension)` | Sets the touch dimension of checkbox.
```
Comment (moog16) [inline packages/mdc-checkbox/README.md]: of the checkbox.

File: packages/mdc-checkbox/_mixins.scss [hunk 2]
```diff
@@ -518,7 +521,7 @@
   $feat-structure: mdc-feature-create-target($query, structure);
 
   @include mdc-feature-targets($feat-structure) {
-    transform: scale(2.75, 2.75);
+    transform: scale(1);
```
Comment (moog16) [inline packages/mdc-checkbox/_mixins.scss]: can you remove scale here? I don't think this is being animated, so you don't need it.

File: test/screenshot/spec/mdc-checkbox/mixins/touch-dimension.html [hunk 3]
```diff
@@ -0,0 +1,164 @@
+<!DOCTYPE html>
+<!--
+  Copyright 2018 Google Inc.
```
Comment (moog16) [inline test/screenshot/spec/mdc-checkbox/mixins/touch-dimension.html]: 2019

File: packages/mdc-checkbox/_mixins.scss [hunk 4]
```diff
@@ -186,6 +186,7 @@
     @include mdc-ripple-surface($query);
     @include mdc-states($mdc-checkbox-baseline-theme-color, $query: $query);
     @include mdc-ripple-radius-unbounded($query: $query);
+    @include mdc-checkbox-touch-dimension($mdc-checkbox-touch-area);
```
Comment (moog16) [inline packages/mdc-checkbox/_mixins.scss]: I need to double check this, but I think this should be outside of .mdc-checkbox

File: test/screenshot/spec/mdc-checkbox/fixture.js [hunk 5]
```diff
@@ -24,4 +24,10 @@
 document.getElementById('checkbox-indeterminate').indeterminate = true;
 document.getElementById('checkbox-indeterminate-disabled').indeterminate = true;
 
+// import {MDCCheckbox} from '../../../../packages/mdc-checkbox';
```
Comment (moog16) [inline test/screenshot/spec/mdc-checkbox/fixture.js]: remove comment.

File: test/screenshot/spec/mdc-checkbox/classes/baseline.html [hunk 6]
```diff
@@ -64,7 +64,7 @@
             <input type="checkbox"
                    checked
                    class="mdc-checkbox__native-control"
-                   id="checkbox-checked"/>
+                   id="checkbox-checked" autofocus="autofocus" />
```
Comment (moog16) [inline test/screenshot/spec/mdc-checkbox/classes/baseline.html]: You don't need the "autofocus". This can just be

`<input autofocus ... />`

**Comment by abhiomkar** [comment]
We do not seems to have polyfill to support for..of syntax in IE11 which cases it to break in IE11. Changed it to use `[].slice.call` to loop checkbox DOM elements.

**Pull request merged**